### PR TITLE
permissions: disable edit and delete buttons for librarians

### LINKED ISSUE #488: A librarian should not be able to edit libraries he/she's not affiliated to.
A librarian is able to edit libraries he/she is not affiliated to. He/She should be able to edit only his/her own library.
Only system librarian can edit all libraries.

### PR DESCRIPTION
**Pull request by BadrAly**
* Creates separate permission files for patron, library, location resources.
* Disables location/library edit and delete buttons for librarians from external libraries.
* Creates complete units tests for library and location permissions.
* Removes duplicates imports.
* Closes #488

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## How to test?

- bootstrap
- login as a `librarian` and make sure you can only edit  your library and its locations
- you should not be able to edit or delete library(locations) of other libraries
- re-login as `system_librarian` and make sure you can edit, delete any library or location of your organisation.

## Code review check list

- [x] Commit message template compliance.
- [x] Commit message without typos.
- [x] File names.
- [x] Functions names.
- [x] Functions docstrings.
- [x] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/locations/permissions.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Location permissions."""
+
+
+from ...permissions import staffer_is_authenticated
+
+
+def can_update_delete_location_factory(record, *args, **kwargs):
```
Comment (jma) [inline rero_ils/modules/locations/permissions.py]: seems the same as the `can_update_delete_location_factory`. Can you use the same permission factory?

File: rero_ils/permissions.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 def user_is_authenticated(user=None):
     """Checks if user is authenticated.
 
-    returns True if user is logged in and authenticated.
+    returns True if user is logged in and authenticated
```
Comment (jma) [inline rero_ils/permissions.py]: Why did you remove the punctuation?

File: tests/api/test_libraries_rest.py [hunk 3]
```diff
@@ -303,7 +303,7 @@ def test_library_secure_api_update(client, lib_fully,
         data=json.dumps(data),
         headers=json_header
     )
-    assert res.status_code == 200
```
Comment (jma) [inline tests/api/test_libraries_rest.py]: For each test can you put some comment why it is not possible? Can you also add a sucess full test: a system librarian modify a library. Can you do the same for the other tests of this commit?

File: ui/src/app/records/search/brief-view/libraries-brief-view.component.ts [hunk 4]
```diff
@@ -93,11 +93,11 @@ import { ToastrService } from 'ngx-toastr';
     <div id="collapseBasic" [collapse]="isCollapsed">
       <ul *ngIf="locations.length" class="list-group list-group-flush">
         <li *ngFor="let location of locations;" class="list-group-item p-1">{{ location.metadata.name }}
-          <a (click)="deleteLocation(location.metadata.pid)"
```
Comment (jma) [inline ui/src/app/records/search/brief-view/libraries-brief-view.component.ts]: Do not forget to do the same in the new `rero-ils-ui` project.

**Review by jma — CHANGES_REQUESTED** [review]

**Review by lauren-d — COMMENTED** [review]
As librarian, I can access directly (by url) to the edit form of another library. Is it possible to protect that?
note: the save action doesn't work as expected

File: ui/src/app/records/search/brief-view/libraries-brief-view.component.ts (hunk 4 above)
Comment (BadrAly) [reply ui/src/app/records/search/brief-view/libraries-brief-view.component.ts]: will create a new PR in the `rero-ils-ui` project

**Comment by BadrAly** [comment]
> As librarian, I can access directly (by url) to the edit form of another library. Is it possible to protect that?
> note: the save action doesn't work as expected

As per @jma, I am opening an issue for this. it is a bug in the `rero-ils-ui` custom-editor.

**Review by lauren-d — APPROVED** [review]

**Review by NicolasLabat — APPROVED** [review]

**Review by pronguen — APPROVED** [review]

**Review by jma — APPROVED** [review]

**Review by rerodak — APPROVED** [review]
